src/Configuration/RectorConfigBuilder.php

```diff
@@ -893,11 +907,26 @@ public function withTypeCoverageLevel(int $level): self
     {
         $this->isTypeCoverageLevelUsed = true;
 
-        $levelRules = LevelRulesResolver::resolve(
-            $level,
-            TypeDeclarationLevel::RULES,
-            'RectorConfig::withTypeCoverageLevel()'
-        );
+        $levelRules = LevelRulesResolver::resolve($level, TypeDeclarationLevel::RULES, __METHOD__);
+
+        $this->rules = array_merge($this->rules, $levelRules);
+
+        return $this;
+    }
+
+    /**
+     * @experimental Since 1.2.5 Raise your PHP level from, one level at a time
+     */
+    public function withPhpLevel(int $level): self
+    {
+        $this->isWithPhpLevelUsed = true;
+
+        $phpLevelSetFilePath = $this->resolvePhpSetsFromComposerJsonPhpVersion();
+
+        $setRectorsResolver = new SetRectorsResolver();
+        $phpRectorRules = $setRectorsResolver->resolveFromFilePathForPhpLevel($phpLevelSetFilePath);
+
+        $levelRules = LevelRulesResolver::resolve($level, $phpRectorRules, __METHOD__);
 
         $this->rules = array_merge($this->rules, $levelRules);
```

Comment:
rules with configuration needs to be append after, like `withCodeQualityLevel()` or other, otherwise, the result will be different, as now not included

https://github.com/rectorphp/rector-src/blob/b2b18abd7e5896611ba96130bcc8372d7e91a784/config/set/php72.php#L18-L29